File: Python/String-2/double_char.py
"""
Given a string, return a string where for every char in the original, there are two chars.

double_char('The') → 'TThhee'
double_char('AAbb') → 'AAAAbbbb'
double_char('Hi-There') → 'HHii--TThheerree'

@author unobatbayar
"""

def double_char(str):
  new = ""
  for i in range(len(str)):
    new = new + str[i]*2
  return new

File: Java/String-2/doubleChar.java
/**

Given a string, return a string where for every char in the original, there are two chars.


doubleChar("The") → "TThhee"
doubleChar("AAbb") → "AAAAbbbb"
doubleChar("Hi-There") → "HHii--TThheerree"

@author unobatbayar
**/

public String doubleChar(String str) {
  String result = "";
  for(int i = 0; i<str.length(); i++) result += str.substring(i, i+1) + str.substring(i, i+1);
  
  return result;
}
